### PR TITLE
Copy utility module to generated module to remove dependency

### PR DESCRIPTION
**Pull request by brywang-msft**
Fixes:
- Enable client-side filtering by default
- Fix Remove-AuthenticodeSignatureBlock

Features:
- -CopyUtilityModuleToOutput copies PSSwaggerUtility to generated module
- -AddUtilityDependencies ensures all dependencies are accessible from within the generated module

Issues: #388

### CODE REVIEW

**Comment by edyoung** [comment]
I don't see any test coverage of the new flags, would be nice to validate in the test suite. Also, do we really need to support both modes (external vs embedded utilities). What would be broken if we only support embedded?

**Comment by brywang-msft** [comment]
Haven't found a way to automate the test yet. Thought I'd make a PR for it later once I figured it out, but I can hold off merging this until a test is added.

As for not supporting external utility module, we'd lose the ability to update the utility module separate from every generated module ever. The embedded module also requires the generated module owner to sign something that's not theirs, which might make certain module owners nervous. In addition, the utility module is already published, which makes it difficult to justify not supporting it.

File: PSSwagger/PSSwagger.Resources.psd1
Comment (deathly809) [inline PSSwagger/PSSwagger.Resources.psd1]: Why three periods?

Comment (brywang-msft) [reply PSSwagger/PSSwagger.Resources.psd1]: Cause three periods == an ellipsis, which I use to indicate things are happening (rather than a note or something has happened). I do the same thing when a menu item opens another menu item or dialog rather than actually doing anything (e.g. "File..." -> { "Open...", "Save" }, "Paste", "Cut")

Will remove the three periods.

Comment (deathly809) [reply PSSwagger/PSSwagger.Resources.psd1]: I was not sure what it meant, other actions don't seem to have it (i.e SkippingExistingParameter)

File: PSSwagger/PSSwagger.psm1
Comment (bmanikm) [inline PSSwagger/PSSwagger.psm1:827]: >Get-Module PSSwaggerUtility [](start = 29, length = 27)

Since PSSwaggerUtility module is a nested module in PSSwagger.psd1, this will not return the expected module info unless user explicitly imports it. #Closed

File: PSSwagger/PSSwagger.psm1
Comment (bmanikm) [inline PSSwagger/PSSwagger.psm1]: >PSSwaggerClientTracing_Dummy.psm1 [](start = 75, length = 33)

Where is this file PSSwaggerClientTracing_Dummy.psm1 getting created?
 #Closed

File: PSSwagger/PSSwagger.psm1
Comment (bmanikm) [inline PSSwagger/PSSwagger.psm1]: Please confirm whether this switch is required for AzureStack modules.

File: PSSwagger/PSSwagger.psm1
Comment (bmanikm) [inline PSSwagger/PSSwagger.psm1]: >$content[$namespaceIndex.LineNumber-1] = "    CSharpNamespace=$($SwaggerDict['info'].NameSpace)" [](start = 20, length = 96)

Please validate the -AsJob functionality with this change. #Closed

**Comment by bmanikm** [comment]
>     public class PSSwaggerClientTracingBase : IServiceClientTracingInterceptor

I see, PSSwaggerJob related classes are not separated from this file. Are there any dependencies required for compiling this on end-user machine for the AzureStack scenario?

---
Refers to: PSSwagger/PSSwaggerUtility/PSSwaggerNetUtilities.Code.ps1:673 in 9296cd1. [](commit_id = 9296cd1ed31e844e2108831b479dbbfafc18f294, deletion_comment = False)

File: PSSwagger/PSSwagger.psm1
Comment (brywang-msft) [reply PSSwagger/PSSwagger.psm1:827]: In what case is the expected module not returned? I tried having the utility module installed globally and not, and in both cases the expected module was returned. #Closed

File: PSSwagger/PSSwagger.psm1
Comment (brywang-msft) [reply PSSwagger/PSSwagger.psm1]: Forgot to add, will add #Closed

**Comment by brywang-msft** [comment]
@bmanikm Yes. You can tell because of the base class. I see you still want me to separate the code. I will do so.

File: PSSwagger/PSSwagger.psm1
Comment (brywang-msft) [reply PSSwagger/PSSwagger.psm1]: Yes. In fact, this switch is the whole point of this change.

**Comment by bmanikm** [comment]
@brywang-msft  As per https://github.com/PowerShell/PSSwagger/issues/388#issuecomment-362024120, AzureStack team doesn't want to include any additional dlls in the generated module for -AsJob functionality.

//Cc: @deathly809

**Comment by brywang-msft** [comment]
@bmanikm The concern was for dlls they would have to sign before distribution. These are external dependencies that come from Azure NuGet packages and would not have to be touched. Either way I have mentioned I will be working on separating the New-PSSwaggerJob code and disabling the rest of the functionality, if possible. **To be clear, this means the external dependencies switch and dlls will be removed.**

**Comment by brywang-msft** [comment]
Update: having trouble with manual AsJob validation, will update again when I figure out what I'm doing wrong.

**Comment by brywang-msft** [comment]
-AsJob has been validated with Cosmos DB.

**Review by bmanikm — APPROVED** [review]
:shipit:

**Pull request merged**
